katip/src/Katip/Core.hs

```diff
@@ -811,12 +811,12 @@ closeScribes le = do
 -- of the log env that are reverted when the supplied monad
 -- completes. 'katipNoLogging', for example, uses this to temporarily
 -- pause log outputs.
-class MonadIO m => Katip m where
+class Katip m where
```

Comment:
This should really be

```suggestion
class Monad m => Katip m where
```

then you won't have to include these `Monad` constraints below.

Reply:
@arybczak is that preferable though? I thought part of the idea of this change was to not require unused constraints. It makes sense to me if an implementation requires Monad that it should demand it but the base class doesn't seem to demand it. Maybe I'm missing something.

Reply:
> I thought part of the idea of this change was to not require unused constraints

The problem isn't really unused constraints, it's just that you're forcing `MonadIO` on downstream users of `Katip` and `MonadIO` is *huge* (in a sense of what you can do with it). `Monad` on the other hand isn't, so including it as a superclass allows you to omit writing `(Katip m, Monad m)` in type signatures (which is somewhat annoying) and doesn't give any practical downsides.